libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -3891,40 +4175,48 @@ CK_DECLARE_FUNCTION( CK_RV, C_VerifyInit )( CK_SESSION_HANDLE hSession,
         }
         else
         {
+            LogError( ( "Failed to initialize verify operation. Could not "
+                        "take xVerifyMutex." ) );
             xResult = CKR_CANT_LOCK;
         }
     }
 
     /* Check that the mechanism and key type are compatible, supported. */
     if( xResult == CKR_OK )
     {
-        xKeyType = mbedtls_pk_get_type( &pxSession->xSignKey );
+        xKeyType = mbedtls_pk_get_type( &pxSession->xVerifyKey );
```

Comment:
Is this change intentional?

Reply:
Yes I stumbled on this bug when doing these logging updates.

Reply:
Good catch – I would search this file for the same/similar error if you haven't already.